src/config_manager.sh

```diff
@@ -287,11 +287,11 @@ function get_config_from_defconfig()
     return 125 # ECANCELED
   fi
 
-  cross_compile=${configurations[cross_compile]}
-  arch=${configurations[arch]}
+  cross_compile=${build_config[cross_compile]}
+  arch=${build_config[arch]}
   # TODO: Right now, the below variable has no meaning and will be empty. Let's
   # expand the config file to enable developers to define ASIC targets.
-  asic=${configurations[asic]}
+  asic=${build_config[asic]}
```

Comment:
tbh, I don't remember why I added it, and this is an excellent example of why we should open issues haha. Let's drop these lines.